### DISCUSSION TITLE
Is it safe to run multiple Wasmer instances in one process

### DISCUSSION THREAD

**Discussion by webmaster128** [wasmerio/wasmer#4260] [Q&A] [answered]
I have a project in which I would like to run multiple independent Wasmer instanes (`Engine` + `Module`s + `Instances`s) at the same time.

I'm asking because the trap handlers that are installed.

- Is this a safe thing to do or are there any process level things happening that can only be done once or can be conflicting?
- Is it safe to run two different versions of Wasmer in one process?

**Comment by syrusakbary** [comment, marked as answer]
> Is this a safe thing to do or are there any process level things happening that can only be done once or can be conflicting?

It should be safe

> Is it safe to run two different versions of Wasmer in one process?

No, the trap handlers can (and probably will) conflict
